lib/samples/validate_utility_network_topology/validate_utility_network_topology.dart

```diff
@@ -0,0 +1,700 @@
+    final result = await job.run();
+
+    // Update the UI with the validatin result.
+    setState(() {
+      _statusTitle = 'Utility Validation Result:';
+      _statusDetail =
+          "Has Dirty Areas: ${result.hasDirtyAreas}\nHas Errors: ${result.hasErrors}\nTap 'Get State' to check the updated network state.";
+      _utilityNetworkCanValidate = result.hasDirtyAreas;
+      _ready = true;
+    });
+  }
+
+  Future<void> performTrace() async {
+    setState(() {
+      _ready = false;
+      _statusTitle = 'Running a downstream trace...';
+      _statusDetail = '';
+    });
+
+    // Clear previous selection from the layers.
+    clearSelectionsFromAllLayers();
+
+    // Get the trace result from the utility network.
+    try {
+      final traceResult = await _utilityNetwork.trace(_traceParameters);
+      final elementTraceResult = traceResult
+          .whereType<UtilityElementTraceResult>()
+          .firstOrNull;
+      final elementsCount = elementTraceResult?.elements.length ?? 0;
+
+      // Select any identified elements in the map.
+      if (elementTraceResult != null) {
+        for (final layer
+            in _mapViewController.arcGISMap!.operationalLayers
+                .whereType<FeatureLayer>()) {
+          final elements = elementTraceResult.elements
+              .where(
+                (element) =>
+                    element.networkSource.featureTable == layer.featureTable,
+              )
+              .toList();
+          final features = await _utilityNetwork.getFeaturesForElements(
+            elements,
+          );
+          layer.selectFeatures(features);
+        }
+      }
+      // Update the state with the trace results.
+      setState(() {
+        _statusTitle = 'Trace completed:';
+        _statusDetail = '$elementsCount elements found';
+        _clearEnabled = true;
+        _ready = true;
+      });
+    } on ArcGISException catch (e) {
+      // If the trace fails update the status.
+      showMessageDialog(e.additionalMessage, title: e.message, showOK: true);
+      setState(() {
+        _statusTitle = 'Trace failed:';
+        _statusDetail = "Tap 'Get State' to check the updated network state.";
+        _clearEnabled = false;
+        _ready = true;
+      });
+    }
+  }
+
+  Future<void> applyEdits() async {
+    if (_selectedFeature == null) return;
+    // Get the service geodatabase and field name of the feature being edited.
+    final table = _selectedFeature!.featureTable as ServiceFeatureTable?;
+    if (table == null || table.serviceGeodatabase == null) return;
+    final serviceGeodatabase = table.serviceGeodatabase!;
+    final fieldName = _currentField?.name;
+    if (fieldName == null) return;
+    setState(() {
+      _statusTitle = 'Updating feature...';
+      _statusDetail = '';
+      _ready = false;
+    });
+    // Set the selected value to the feature.
+    _selectedFeature!.attributes[fieldName] = _selectedCodedValue?.code;
```

Comment:
If `_selectedCodedValue` is null, we wouldn't want to make this assignment, I don't think. We'd want to bail out with an error condition. Maybe check `_selectedCodedValue` up above, where `_selectedFeature` is checked.